R/position-nudge.R

```diff
@@ -27,10 +27,23 @@
 #'   geom_point() +
 #'   geom_text(aes(label = y), nudge_y = -0.1)
 position_nudge <- function(x = 0, y = 0) {
-  ggproto(NULL, PositionNudge,
-    x = x,
-    y = y
-  )
+  # find the appropriate position adjustment for the given x and y settings;
+  # not nudging in dimensions that are not required reduces the risk
+  # spurious breaks, e.g. when the geom only has an x coordinate but no
+  # y coordinate
+  if (x != 0) {
+    if (y != 0) {
+      pos <- ggproto(NULL, PositionNudge, x = x, y = y)
```

Comment:
Why do these need to be separate objects? Couldn't the choice (i.e. this if-else block) be placed in `PositionNudge$compute_layer()`?

Reply:
The primary difference among these objects is the required aes. If we want a single nudge object then it can't have any required aes. I don't have a strong opinion about it one way or another.

Reply:
Seems reasonable to me to just drop the required aes.

Reply:
Done.